torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -114,6 +118,9 @@ def __init__(
     ):
         self.modules = modules  # permit ref cycle because 1:1 lifetime
         param_module_infos = _get_param_module_infos(params, modules)
+        if device is None:
+            raise RuntimeError("Expects torch.device, got None")
+
```

Comment:
super nit: let us remove this just to de-clutter because `device` is already annotated as `torch.device`, not `Optional[torch.device]` or something

I think it was just a small oversight in the compile test before. Anyways, we would error on L139 if trying to access `None.type`.

Reply:
@awgu removed the none check as you mentioned it will anyway fail in L139.